lib/ansible/modules/network/aci/msc_role.py

```diff
@@ -0,0 +1,210 @@
+    description: North European Datacenter
+    state: present
+  delegate_to: localhost
+
+- name: Remove a role
+  msc_role:
+    host: msc_host
+    username: admin
+    password: SomeSecretPassword
+    name: north_europe
+    state: absent
+  delegate_to: localhost
+
+- name: Query a role
+  msc_role:
+    host: msc_host
+    username: admin
+    password: SomeSecretPassword
+    name: north_europe
+    state: query
+  delegate_to: localhost
+  register: query_result
+
+- name: Query all roles
+  msc_role:
+    host: msc_host
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+  register: query_result
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.aci.msc import MSCModule, msc_argument_spec, issubset
+
+
+def main():
+    argument_spec = msc_argument_spec()
+    argument_spec.update(
+        role=dict(type='str', required=False, aliases=['name', 'role_name']),
+        role_id=dict(type='str', required=False),
+        display_name=dict(type='str'),
+        description=dict(type='str'),
+        permissions=dict(type='list', choices=[
+            'manage-roles',
+            'manage-schemas',
+            'manage-sites',
+            'manage-tenants',
+            'manage-users',
+            'view-roles',
+            'view-schemas',
+            'view-sites',
+            'view-tenants',
+            'view-tenant-schemas',
+            'view-users',
+        ]),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['role']],
+            ['state', 'present', ['role']],
+        ],
+    )
+
+    role = module.params['role']
+    role_id = module.params['role_id']
+    description = module.params['description']
+    permissions = module.params['permissions']
+    state = module.params['state']
+
+    msc = MSCModule(module)
+
+    path = 'roles'
+
+    # Query for existing object(s)
+    if role_id is None and role is None:
+        msc.existing = msc.query_objs(path)
+    elif role_id is None:
+        msc.existing = msc.get_obj(path, name=role)
+        if msc.existing:
+            role_id = msc.existing['id']
+    elif role is None:
+        msc.existing = msc.get_obj(path, id=role_id)
+    else:
+        msc.existing = msc.get_obj(path, id=role_id)
+        existing_by_name = msc.get_obj(path, name=role)
+        if existing_by_name and role_id != existing_by_name['id']:
+            msc.fail_json(msg="Provided role '{1}' with id '{2}' does not match existing id '{3}'.".format(role, role_id, existing_by_name['id']))
+
+    # If we found an existing object, continue with it
+    if role_id:
+        path = 'roles/{id}'.format(id=role_id)
+
+    if state == 'query':
+        pass
+
+    elif state == 'absent':
```

Comment:
Same as others...should msc.changed be updated?

Reply:
Taken care of by `exit_json()` and `fail_json()`.